app/domain/operations/hbx_enrollments/drop_enrollment_members.rb

```diff
@@ -52,10 +55,45 @@ def drop_enrollment_members(params)
         Success(dropped_member_info)
       end
 
+      def remove_dropped_enr_members(new_enrollment, non_eligible_members)
+        ineligible_applicant_ids = non_eligible_members.pluck(:applicant_id)
+        new_enrollment.hbx_enrollment_members.where(:applicant_id.in => ineligible_applicant_ids).delete
+      end
+
+      def terminate_base_enrollment(base_enrollment, new_enrollment, terminate_date)
+        return if  base_enrollment.coverage_expired?
+
+        if base_enrollment.may_terminate_coverage? && (new_enrollment.effective_on > base_enrollment.effective_on)
+          base_enrollment.terminate_coverage!
+          base_enrollment.update_attributes!(terminated_on: terminate_date)
+        elsif base_enrollment.may_cancel_coverage?
+          base_enrollment.cancel_coverage!
+        end
+      end
+
+      def ivl_params(base_enrollment, new_effective_date)
+        { product_id: product_id_csr_variant(base_enrollment, new_effective_date.year),
+          consumer_role_id: base_enrollment.consumer_role_id,
+          predecessor_enrollment_id: base_enrollment.id,
+          rating_area_id: ::BenefitMarkets::Locations::RatingArea.rating_area_for(base_enrollment.consumer_role.rating_address, during: new_effective_date)&.id }
+      end
+
+      def product_id_csr_variant(base_enrollment, effective_year)
+        tax_household = base_enrollment.family.active_household.latest_active_thh_with_year(effective_year)
+        base_enrollment_product_id = base_enrollment.product_id
+        return base_enrollment_product_id unless tax_household.present?
+
+        eligible_csr = tax_household.eligibile_csr_kind(base_enrollment.hbx_enrollment_members.map(&:applicant_id))
+        csr_variant = EligibilityDetermination::CSR_KIND_TO_PLAN_VARIANT_MAP[eligible_csr]
+        products = ::BenefitMarkets::Products::HealthProducts::HealthProduct.by_year(effective_year).where({:hios_id => "#{base_enrollment.product.hios_base_id}-#{csr_variant}"})
+        products.count >= 1 ? products.last.id : base_enrollment_product_id
+      end
```

Comment:
Check dependent drop enrollment product available in current consumer service area, if not available return false saying "Product not available in current rating area"

 ::Operations::Products::ProductOfferedInServiceArea.new.call({enrollment: drop_enrollment})

Reply:
Copied this from reinstatement as suggested. I will check with the dev who worked on this.

Reply:
Any time that involves rating area change we need to check product available in new service area or not ..if above operation returns failure we should not create dep drop enrollment.

Reply:
Will return failure if the product is not available.